src/guide/chapters/core-language.md

```diff
@@ -176,39 +177,39 @@ A record is a set of key-value pairs, similar to objects in JavaScript or Python
 
 ```elm
 > point = { x = 3, y = 4 }
-{ x = 3, y = 4 }
+{ x = 3, y = 4 } : { x : number, y : number' }
 
 > point.x
-3
+3 : number
 
 > bill = { name = "Gates", age = 57 }
-{ age = 57, name = "Gates" }
+{ age = 57, name = "Gates" } : { age: number, name: String }
 
 > bill.name
-"Gates"
+"Gates" : String
 ```
 
 So we can create records using curly braces and access fields using a dot. Elm also has a version of record access that works like a function. By starting the variable with a dot, you are saying please access the field with the following name, so `.name` accesses the `name` field of the record.
 
 ```elm
 > .name bill
-"Gates"
+"Gates" : String
 
 > List.map .name [bill,bill,bill]
-["Gates","Gates","Gates"]
+["Gates","Gates","Gates"] : List String
 ```
 
 When it comes to making functions with records, you can do some pattern matching to make things a bit lighter.
 
 ```elm
 > under70 {age} = age < 70
-<function> 
+<function> : { a | age : number } -> Bool
```

Comment:
Besides documenting the simplest cases (`String`, `number`, etc.), here's one case where the newly displayed type annotations are probably going to help a lot (pattern matching). Another case for updating the guide with the latest REPL's output :monkey_face: